### PR TITLE
bugfix: CLDSRV-5 Remove extra headers from 304 responses

### PR DESCRIPTION
**Pull request by miniscruff**
Note: This will fail until the arsenal version is updated from https://github.com/scality/Arsenal/pull/1526

### CODE REVIEW

**Review by naren-rajendran — APPROVED** [review]

**Review by saitharunthodupunuru — APPROVED** [review]

File: tests/functional/aws-node-sdk/test/object/get.js [hunk 1]
```diff
@@ -658,13 +658,19 @@ describe('GET object', () => {
...
+                        Bucket: bucketName,
+                        Key: objectName,
                         IfNoneMatch: etagTrim,
                         IfModifiedSince: dateFromNow(-1),
                     }, err => {
                         checkError(err, 'NotModified');
                         done();
                     });
+                    req.on('httpHeaders', (code, headers, response, status) => {
+                        assert(!headers.hasOwnProperty('content-type'));
```
Comment (saitharunthodupunuru) [inline tests/functional/aws-node-sdk/test/object/get.js]: Change this to

```suggestion
                       assert(!headers.hasOwnProperty('content-type') || !headers.hasOwnProperty('Content-Type'));
```

File: tests/functional/aws-node-sdk/test/object/get.js [hunk 2]
```diff
@@ -658,13 +658,19 @@ describe('GET object', () => {
...
+                        assert(!headers.hasOwnProperty('content-length'));
```
Comment (saitharunthodupunuru) [inline tests/functional/aws-node-sdk/test/object/get.js]: same as above

**Review by saitharunthodupunuru — CHANGES_REQUESTED** [review]

File: tests/functional/aws-node-sdk/test/object/get.js (hunk 1 above)
Comment (miniscruff) [reply tests/functional/aws-node-sdk/test/object/get.js]: Hmm, I know from the outputs the results have all lower case keys so not sure it is worth adding the other case.

Comment (saitharunthodupunuru) [reply tests/functional/aws-node-sdk/test/object/get.js]: Hmm That is fine too. If the output has always lower case

**Review by saitharunthodupunuru — APPROVED** [review]

File: tests/functional/aws-node-sdk/test/object/get.js (hunk 1 above)
Comment (rahulreddy) [reply tests/functional/aws-node-sdk/test/object/get.js]: No need to check for case sensitiveness. Node.js converts all headers to lower case https://nodejs.org/api/http.html#http_message_headers
Also a check like `!headers['content-type']` is sufficient.

**Pull request merged**